linux/systemd/ckb-next-daemon.service.in

```diff
@@ -7,6 +7,11 @@ Description=Corsair Keyboards and Mice Daemon
 [Service]
 ExecStart=@CMAKE_INSTALL_PREFIX@/bin/ckb-next-daemon
 Restart=on-failure
+ProtectSystem=full
+ProtectHome=True
+PrivateNetwork=True
+PrivateTmp=True
```

Comment:
This will break firmware updates

Reply:
Is it because it needs to be able to have full write access the ckb-next-daemon? Is it one of these aspects in particular that would cause it to break?

I can adjust things such that ProtectHome is 'read-only' if that would make things easier.

Also with ProtectSystem only /usr, /etc and /boot are read-only. Does the daemon need write access here too?

Reply:
I removed the privateTMP.

Could I perhaps know what happens with the firmware update process if possible.

I assume a file is downloaded add to tmp, then ckb-next-daemon reads it and writes it to some file descriptor? Is that the general jist?

I was also experimenting a little bit with other services like ufw and wpa_supplicant, i will see if there are some further changes if possible and get back to this.

Reply:
I edited this to add a few more permissions that I found I was able to apply with other services. I will test this further when I next get access to my keyboard tomorrow.

Reply:
So most of the above seems to work. I had some issues with the RestrictSUIDSGID which kind of broke changing the colours on the keyboard.

Reply:
I also found out where to find which systemcalls are allowed.

Basically run 'systemd-analyze syscall-filter'

Then filter on system-service.

Reply:
I seemed to have issues with the system-service filter list. Especially on restart where it would throw an error. I'll take another look later.